### Imports :

In [1]:
from keras.models import Sequential
from keras.layers import Dense
import pandas as pd
import numpy as np

### Model :

In [2]:
def regression_model(n_cols):
    model = Sequential()
    model.add(Dense(10,activation='relu', input_shape=(n_cols,)))   # 10 nodes in hidden layer
    model.add(Dense(1)) # output layer
    
    model.compile(optimizer='adam', loss='mean_squared_error')
    return model

### Dataset :

In [3]:
concrete_data = pd.read_csv('https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DL0101EN/labs/data/concrete_data.csv')

In [4]:
predictors = concrete_data.loc[:, concrete_data.columns!='Strength']
target = concrete_data['Strength']

In [5]:
predictors_norm = (predictors-predictors.mean())/predictors.std()

### Training and Testing model :

In [6]:
model = regression_model(predictors_norm.shape[1])

In [7]:
mse_list = []

for i in range(50):
    print(f"current iteration : {i+1}")
    history = model.fit(predictors_norm, target, validation_split=0.3, epochs=100, verbose=0, shuffle=True) # 100 epochs
    final_loss = history.history['val_loss'][-1]
    mse_list.append(final_loss)


current iteration : 1
current iteration : 2
current iteration : 3
current iteration : 4
current iteration : 5
current iteration : 6
current iteration : 7
current iteration : 8
current iteration : 9
current iteration : 10
current iteration : 11
current iteration : 12
current iteration : 13
current iteration : 14
current iteration : 15
current iteration : 16
current iteration : 17
current iteration : 18
current iteration : 19
current iteration : 20
current iteration : 21
current iteration : 22
current iteration : 23
current iteration : 24
current iteration : 25
current iteration : 26
current iteration : 27
current iteration : 28
current iteration : 29
current iteration : 30
current iteration : 31
current iteration : 32
current iteration : 33
current iteration : 34
current iteration : 35
current iteration : 36
current iteration : 37
current iteration : 38
current iteration : 39
current iteration : 40
current iteration : 41
current iteration : 42
current iteration : 43
current iteration : 

In [8]:
mean = np.array(mse_list).mean()
std = np.array(mse_list).std()

In [9]:
print(f"mse_list mean is {mean:.3f} and standard deviation is {std:.3f}")

mse_list mean is 105.940 and standard deviation is 23.054


#### comparison

||B|C|Difference|
|-|-|-|-|
|mean(mse)|119.625|105.940|13.685|

Note : results may vary due to shuffling